traitlets/traitlets.py

```diff
@@ -640,36 +641,69 @@ class dict to the newly created class ``cls``.
 
 
 def observe(*names):
-    """ A decorator which can be used to observe members on a class.
+    """ A decorator which can be used to observe Traits on a class.
 
     Parameters
     ----------
     *names
-        The str names of the attributes to observe on the object.
+        The str names of the Traits to observe on the object.
     """
     return ObserveHandler(names)
 
+def validate(name):
+    """ A decorator which validates a HasTraits object's state when a Trait is set.
+
+    Parameters
+    ----------
+    name
+        The str name of the Trait to observe on the object.
+    """
+    return ValidateHandler(name)
+    
 
-class ObserveHandler(BaseDescriptor):
 
-    def __init__(self, names=None):
-        if names is None:
-            self.names=[None]
-        else:
-            self.names = names
+class EventHandler(BaseDescriptor):
 
-    def __call__(self, func):
+    def _init_call(self, func):
         self.func = func
         return self
 
-    def __get__(self, inst, cls):
+    def _func_call(self, inst, *args, **kwargs):
+        method = types.MethodType(self.func, inst)
+        return method(*args, **kwargs)
+
+    def __call__(self, *args, **kwargs):
```

Comment:
Old `__call__` method forced the use of `super` to access overwritten methods. 9ea2afa enables this syntax:

```python
from traitlets import HasTraits, Int, observe

class A(HasTraits):
    i, j = Int(), Int()
    @observe('i')
    def _i_changed(self, change):
        self.j = change['new']
        
    
class B(A):
    k = Int()
    @observe('i')
    def _i_changed(self, change):
        A._i_changed(self, change)
        self.k = change['new']
```